nasa/ow_simulator
# Oceanwater 1015 arm end effector force torque telemetry

## Linked Issues:
| EPIC ⚡ | [OCEANWATER-1015](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-1015) |
| :----------- | :----------- |
| Jira Ticket 🎟️ | [OCEANWATER-1015](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-1015) |
| Github :octocat:  | # |


## Summary of Changes
Update Force/Torque sensor telemetry to unified standard
* update topic `/ft_sensor_dist_pitch` topic to `/arm_end_effector_force_torque`
* add `ArmEndEffectorForceTorque.msg` to the owl_msgs package with updated semantics
* update code to use float array of new msg semantics over previous `geometry_msg/WrenchStamped` msg
* update fault injection to map gazebo sensor topic to new topic name
* change variable names to be consistent with new naming scheme

## Test
*Ensure the ow_autonomy repo is on the [934 epic branch](https://github.com/nasa/ow_autonomy/tree/OCEANWATER-934_telemetry_unification_epic)*

Test that the sensor information is published on the correct topic
* Open two sourced terminals, in one launch the sim `roslaunch ow atacamy_y1a.launch`
* In the second terminal, monitor the newly renamed topic `rostopic echo /arm_end_effector_force_torque`
* Observe the force torque values displaying rapidly

Test that fault injection redirects the topic and behaves correctly
* Add the FT sensor faults for zero_signal_failure and enable. Observe the values zero out.
* Uncheck the zero_signal_failure fault and move the signal_bias slider to the right and observe the values changing by ~1.0
* Reset the signal_bias slider and now move the signal_noise_slider to the right and observe an increase in noise in the values.

~~Test the modified ros_test~~
* ~~run `rostest ow_sim_tests ft_sensor.test`~~
* ~~the test should pass, but there is a failure on closure that is present in noetic-devel~~

*Note: The previous test is not working regardless of changes in this branch. However, datatype changes were made to be consistent with msg changes for future work fixing the ros_test.*
